ui/packages/catalog-ui-search/src/main/webapp/extension-points/table-export/table-export.tsx

```diff
@@ -32,7 +39,169 @@ type ExportResponse = {
+  })
+  if (count && queryParts.length > count) {
+    queryParts = queryParts.slice(0, count)
+  }
+  return `(${queryParts.join(' OR ')})`
+}
+const visibleData = (selectionInterface: any) =>
+  buildCqlQueryFromMetacards(
+    selectionInterface.getActiveSearchResults().toJSON(),
+    null
+  )
+const allData = (selectionInterface: any) =>
+  selectionInterface.getCurrentQuery().get('cql')
+function getCqlForSize(exportSize: string, selectionInterface: any) {
+  return exportSize === 'visible'
+    ? visibleData(selectionInterface)
+    : allData(selectionInterface)
+}
+function getSrcs(selectionInterface: any) {
+  const srcs = selectionInterface.getCurrentQuery().get('src')
+  return srcs === undefined ? _.pluck(Sources.toJSON(), 'id') : srcs
+}
+function getColumnOrder(): string[] {
+  return user
+    .get('user')
+    .get('preferences')
+    .get('columnOrder')
+}
+function getHiddenFields(): string[] {
+  return user
+    .get('user')
+    .get('preferences')
+    .get('columnHide')
+}
+function getHits(sources: Source[]): number {
+  return sources
+    .filter(source => source.id !== 'cache')
+    .reduce((hits, source) => (source.hits ? hits + source.hits : hits), 0)
+}
+function getExportCount({
+  exportSize,
+  selectionInterface,
+  customExportCount,
+}: ExportCountInfo): number {
+  if (exportSize === 'custom') {
+    return customExportCount
+  }
+  const result = selectionInterface.getCurrentQuery().get('result')
+  return exportSize === 'all'
+    ? getHits(result.get('status').toJSON())
+    : result.get('results').length
+}
+function getSorts(selectionInterface: any) {
+  return selectionInterface.getCurrentQuery().get('sorts')
+}
+export const getWarning = (exportCountInfo: ExportCountInfo): string => {
+  const exportCount = getExportCount(exportCountInfo)
+  const result = exportCountInfo.selectionInterface
+    .getCurrentQuery()
+    .get('result')
+  const totalHits = getHits(result.get('status').toJSON())
+  const limitWarning = `You cannot export more than the administrator configured limit of ${
+    properties.exportResultLimit
+  }.`
+  let warningMessage = ''
+  if (exportCount > properties.exportResultLimit) {
+    if (exportCountInfo.exportSize === 'custom') {
+      return limitWarning
+    }
+    warningMessage =
+      limitWarning +
+      `  Only ${properties.exportResultLimit} ${
+        properties.exportResultLimit === 1 ? `result` : `results`
+      } will be exported.`
+  }
+  if (exportCountInfo.exportSize === 'custom') {
+    if (exportCount > totalHits) {
+      warningMessage = `You are trying to export ${exportCount} results but there ${
+        totalHits === 1 ? `is` : `are`
+      } only ${totalHits}.  Only ${totalHits} ${
+        totalHits === 1 ? `result` : `results`
+      } will be exported.`
+    }
+  }
+  if (
+    totalHits > 100 &&
+    exportCount > 100 &&
+    properties.exportResultLimit > 100
+  ) {
+    warningMessage += `  This may take a long time.`
```

Comment:
✏️ What's a long time, 2 minutes or 2 hours? Is it possible to give a baseline time estimate? ie `This may take a few minutes` / `This may take several minutes`?

Reply:
Honestly don't think we can make a prediction on the time estimate because the the export limit can be set arbitrarily high by the admin and could be running on a very slow network. Don't think additional logic would be worth the effort, and I just moved the warning message code around, that warning message has been in place so I would prefer not to change it